magenta/models/music_vae/music_vae_export.py

```diff
@@ -0,0 +1,219 @@
+def build_signature_defs(input_tensors, output_tensors):
+    sample_signature_def = build_sample_signature(input_tensors, output_tensors)
+    encode_signature_def = build_encode_signature(input_tensors, output_tensors)
+    decode_signature_def = build_decode_signature(input_tensors, output_tensors)
+
+    signature_def_map = {
+        'sample': sample_signature_def,
+        'encode': encode_signature_def,
+        'decode': decode_signature_def
+    }
+
+    return signature_def_map
+
+
+# TODO create a named registry of signature def builders keyed by name for better type safety
+# Sample seems almost exactly like decode from the trained model code, possibly redundant
+def build_sample_signature(input_tensors, output_tensors):
+    sample_inputs = {
+        'temperature': saved_model.utils.build_tensor_info(input_tensors.temperature),
+        'max_length': saved_model.utils.build_tensor_info(input_tensors.max_length)
+    }
+    if input_tensors.c_input is not None:
+        sample_inputs['c_input'] = saved_model.utils.build_tensor_info(input_tensors.c_input)
+    if input_tensors.z_input is not None:
+        sample_inputs['z_input'] = saved_model.utils.build_tensor_info(input_tensors.z_input)
+
+    sample_outputs = {
+        'outputs': saved_model.utils.build_tensor_info(output_tensors.outputs)
+
+    }
+
+    sample_signature_def = saved_model.signature_def_utils.build_signature_def(
+        inputs=sample_inputs,
+        outputs=sample_outputs,
+        method_name="sample"
+    )
+
+    return sample_signature_def
+
+
+def build_encode_signature(input_tensors, output_tensors):
+    # define the signature for the encoding operation
+    encode_inputs = {
+        'inputs': saved_model.utils.build_tensor_info(input_tensors.inputs),
+        'controls': saved_model.utils.build_tensor_info(input_tensors.controls),
+        'inputs_length': saved_model.utils.build_tensor_info(input_tensors.inputs_length)
+    }
+    encode_outputs = {
+        'mu': saved_model.utils.build_tensor_info(output_tensors.mu),
+        'sigma': saved_model.utils.build_tensor_info(output_tensors.sigma),
+        'z': saved_model.utils.build_tensor_info(output_tensors.z)
+    }
+
+    encode_signature_def = saved_model.signature_def_utils.build_signature_def(
+        inputs=encode_inputs,
+        outputs=encode_outputs,
+        method_name="encode"
+    )
+
+    return encode_signature_def
+
+
+def build_decode_signature(input_tensors, output_tensors):
+    decode_inputs = {
+        'temperature': saved_model.utils.build_tensor_info(input_tensors.temperature),
+        'max_length': saved_model.utils.build_tensor_info(input_tensors.max_length),
+        'z_input': saved_model.utils.build_tensor_info(input_tensors.z_input)
+    }
+    if input_tensors.c_input:
+        decode_inputs["c_input"] = saved_model.utils.build_tensor_info(input_tensors.c_input)
+
+    # Question for magenta team about this, appears in trained model as a return value from decode,
```

Comment:
These are really just used internally for the hierarchical decoder to be able to pass state around.

Reply:
Great, removed